osu.Game/Skinning/LegacySkin.cs

```diff
@@ -142,8 +133,17 @@ protected override void Dispose(bool isDisposing)
 
                     break;
 
+                case LegacySkinConfiguration.LegacySetting legacy:
+                    switch (legacy)
+                    {
+                        case LegacySkinConfiguration.LegacySetting.Version:
+                            return SkinUtils.As<TValue>(new Bindable<decimal>(Configuration.LegacyVersion ?? LegacySkinConfiguration.LATEST_VERSION));
+                    }
+
+                    goto default;
```

Comment:
you replaced one avoided syntax with another.

Reply:
got any alternatives? I can't `break;` because other settings need to be parsed inside the `default` case.

Reply:
i don't follow

Reply:
How can I achieve the same behaviour with a different syntax (i.e. how can I resolve this conversation)?

Reply:
I don't follow what I should be looking at. Can you provide a link to the code where `HitCirclePrefix` is doing this?

Reply:
I would split handling the entire `LegacySkinConfiguration` enum to a switch in a separate method and call that inside of this main switch. That would also require extracting a method for the generic lookup path (so that it can be called in both the main switch and the legacy-specific one).

Reply:
I have pushed in the aforementioned refactoring here. Let me know if it is more to your liking.